src/Illuminate/Session/Middleware/StartSession.php

```diff
@@ -143,13 +144,18 @@ protected function storeCurrentUrl(Request $request, $session)
      */
     protected function collectGarbage(SessionInterface $session)
     {
-        $config = $this->manager->getSessionConfig();
-
-        // Here we will see if this request hits the garbage collection lottery by hitting
-        // the odds needed to perform garbage collection on any given request. If we do
-        // hit it, we'll call this handler to let it delete all the expired sessions.
-        if ($this->configHitsLottery($config)) {
-            $session->getHandler()->gc($this->getSessionLifetimeInSeconds());
+        // Cache based sessions use their own mechanism to invalidate entries
+        // once their lifetime expires so there's no need to perform the garbage
+        // collection for them.
+        if (! $this->sessionUsesCacheBasedHandler($session)) {
```

Comment:
Supports isn't really correct. They both support it, just not all of them need it.